pkg/common/scrubber.go

```diff
@@ -0,0 +1,341 @@
+/*
+Copyright 2023 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/gosecretive"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	ReferencePrefix                  = "$ref:"
+	ReferenceToEnvVarPrefix          = "NUCLIO_B64_"
+	NuclioFlexVolumeSecretNamePrefix = "nuclio-flexvolume"
+	SecretTypeFunctionConfig         = "nuclio.io/functionconfig"
+	SecretTypeV3ioFuse               = "v3io/fuse"
+	SecretContentKey                 = "content"
+	FunctionSecretMountPath          = "/etc/nuclio/secrets"
+)
+
+type Scrubber interface {
+	ValidateReference(objectToScrub interface{},
+		existingSecretMap map[string]string,
+		fieldPath,
+		secretKey,
+		stringValue string) error
+	ConvertMapToConfig(mapConfig interface{}) (interface{}, error)
+}
```

Comment:
Also, add docstrings and new lines between interface functions

Reply:
@TomerShor added scrub and restore, we should be fine with having `Scrub` and `Restore` only in AbstractScrubber anyway, because they all inherit it.